server/sonar-db-dao/src/main/java/org/sonar/db/component/BranchDto.java

```diff
@@ -133,6 +134,27 @@ public BranchDto setMergeBranchUuid(@Nullable String s) {
     return this;
   }
 
+  @Override
+  public boolean equals(Object o) {
```

Comment:
if equals is defined, hashcode should be defined too. Have a look at each method's javadoc.

Reply:
Dude, of course I know about the equals + hashCode contract, what kind of amature do you take me for! My point had nothing to do with breaking that contract. It looked to me you added equals + hashCode just to be able to put instances in sets. I was questioning if that's necessary, because I had another idea in mind that could make these methods unnecessary. I think comparing DTOs (for equality) is a code smell, which implies that these methods are pointless boilerplate, just to support a code smell. But it doesn't matter, as you mentioned in another comment, if this makes testing easier, I'm fine with that.